### PR TITLE
feat: include version 2.1 and new bindings

### LINKED ISSUE #80: Support v2.1.0 of the spec
#### Reason/Context
We need to add v2.1.0 specific example documents.

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR introduces a few changes to the documents.

1. Bindings have been separated completely from the AsyncAPI spec files, this is because the binding versions are, as I understand completely separate from the spec version.
2. Each binding protocol has been given its own version subfolder i.e. instead of `tests/bindings/Message Bindings Object/http/valid.yaml` the path given is `tests/bindings/Message Bindings Object/http/0.1.0/valid.yaml`.
3. I have only added documents for the changes introduces in AsyncAPI 2.1. Not sure how we should handle this moving forward for the upcoming major versions?

**Related issue(s)**
fixes #80

### CODE REVIEW

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
Please refer to my comments, overall is good :)

However I have problem which should be addressed in the next PR. Why we use the names of the folder with spaces? I don't think so that it's a good, because as I remember we have in plan to publish the `tck` as separate package to the npm.

The another problem. Why we have the path format for bindings: `bindings/{OBJECT...}/{BINDING}/{VERSION}`? I think that `bindings/{BINDING}/{VERSION}/{OBJECT...}` will be better.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # AsyncAPI tck
-AsyncAPI's Test Compatibility Kit (TCK) provides a way for AsyncAPI processors to test their level of compliance with the [AsyncAPI 2.0 Spec](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md). This TCK contains a set of AsyncAPI documents meant to be used to test correct and incorrect usage of each AsyncAPI feature.
+AsyncAPI's Test Compatibility Kit (TCK) provides a way for AsyncAPI processors to test their level of compliance with the specification. This TCK contains a set of AsyncAPI documents meant to be used to test correct and incorrect usage of each AsyncAPI feature.
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply README.md:2]: Might be a good idea yea, changing 🙂

File: tests/asyncapi-2.1/Message Object/valid.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+asyncapi: 2.0.0
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply tests/asyncapi-2.1/Message Object/valid.yaml]: Good point, changed

**Comment by jonaslagoni** [comment]
> However I have problem which should be addressed in the next PR. Why do we use the names of the folder with spaces? I don't think so it's a good, because as I remember we have in plan to publish the `tck` as separate package to the npm.

Yea I agree, do you mind creating another issue for it?

> The another problem. Why do we have the path's format for bindings: `bindings/{OBJECT...}/{BINDING}/{VERSION}`? I think that `bindings/{BINDING}/{VERSION}/{OBJECT...}` will be better.

You are right, much cleaner setup, changing 🙂

**Comment by magicmatatjahu** [comment]
> Yea I agree, do you mind creating another issue for it?

https://github.com/asyncapi/tck/issues/86 :)

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
+
+The TCK is build for version <= 2.0.0.
```
Comment (magicmatatjahu) [inline README.md:4]:
```suggestion
The TCK is build for version >=2.0.0.
```

😄

**Review by magicmatatjahu — COMMENTED** [review]
One change 😄

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Comment by jonaslagoni** [comment]
@magicmatatjahu one thing that dawned on me, saying the bindings do not follow the spec version is bullshit 😆 At least it is once we restructure the operation, message, and channels.

Do you think this is the right approach or should bindings be moved into `asyncapi-2.0` ? 🤔

**Comment by magicmatatjahu** [comment]
@jonaslagoni We have issue for specify spec version in the binding -> https://github.com/asyncapi/spec/issues/590 so I think that binding are independent from the spec version. I mean that we should test the binding with asyncapi version separately, so `bindings` folder is good for me :)

**Pull request merged**